tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -56,14 +56,22 @@ def montecarlo_main_loop(packet_collection, numba_model, numba_plasma,
     for i in prange(len(output_nus)):
         if montecarlo_configuration.single_packet_seed != -1:
             i = montecarlo_configuration.single_packet_seed
-        r_packet = RPacket(numba_model.r_inner[0],
-                           packet_collection.packets_input_mu[i],
-                           packet_collection.packets_input_nu[i],
-                           packet_collection.packets_input_energy[i],
-                           i)
-
-        # We want to set the seed correctly per user; otherwise, random.
-        np.random.seed(i)
+            r_packet = RPacket(numba_model.r_inner[0],
+                               packet_collection.packets_input_mu[i],
+                               packet_collection.packets_input_nu[i],
+                               packet_collection.packets_input_energy[i],
+                               i)
+            np.random.seed(i)
+        else:
+            r_packet = RPacket(numba_model.r_inner[0],
```

Comment:
At first glance, this looks identical to lines 59-63. Is there a reason why this cannot be moved outside of the if-else clause?

Reply:
I think you're right — I'm pretty sure this was for debugging purposes a while back. I'll change this!